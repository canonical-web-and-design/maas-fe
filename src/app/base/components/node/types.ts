import type { AnyAction } from "redux";

import type { CommonActionFormProps } from "app/base/types";
import type { Node } from "app/store/types/node";

export type NodeActionFormProps<E = null> = CommonActionFormProps<E> & {
  cleanup?: () => AnyAction;
  nodes: Node[];
  modelName: string;
};
