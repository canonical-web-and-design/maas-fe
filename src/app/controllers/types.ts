import type { ValueOf } from "@canonical/react-components";

import type { SidePanelContent, SetSidePanelContent } from "app/base/types";
import type { ControllerHeaderViews } from "app/controllers/constants";

export type ControllerSidePanelContent = SidePanelContent<
  ValueOf<typeof ControllerHeaderViews>
>;

export type ControllerSetSidePanelContent =
  SetSidePanelContent<ControllerSidePanelContent>;
