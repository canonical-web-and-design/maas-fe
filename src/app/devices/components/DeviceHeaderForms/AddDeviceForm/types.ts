import type {
  CreateParams,
  Device,
  DeviceIpAssignment,
} from "app/store/device/types";
import type { Domain } from "app/store/domain/types";
import type { Zone } from "app/store/zone/types";

type CreateParamsInterface = CreateParams["interfaces"][0];

export type AddDeviceInterface = {
  id: number;
  ip_address: NonNullable<CreateParamsInterface["ip_address"]>;
  ip_assignment: DeviceIpAssignment;
  mac: CreateParamsInterface["mac"];
  name: NonNullable<CreateParamsInterface["name"]>;
  subnet: NonNullable<CreateParamsInterface["subnet"]> | "";
};

export type AddDeviceValues = {
  domain: Domain["name"];
  hostname: Device["hostname"];
  interfaces: AddDeviceInterface[];
  zone: Zone["name"];
};
