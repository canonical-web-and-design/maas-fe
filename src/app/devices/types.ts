import type { ValueOf } from "@canonical/react-components";

import type { SidePanelContent, SetSidePanelContent } from "app/base/types";
import type { DeviceHeaderViews } from "app/devices/constants";

export type DeviceSidePanelContent = SidePanelContent<
  ValueOf<typeof DeviceHeaderViews>
>;

export type DeviceSetSidePanelContent =
  SetSidePanelContent<DeviceSidePanelContent>;
