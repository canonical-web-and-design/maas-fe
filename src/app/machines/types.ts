import type { ValueOf } from "@canonical/react-components";

import type { MachineHeaderViews } from "./constants";

import type { HardwareType } from "app/base/enum";
import type {
  CommonActionFormProps,
  HeaderContent,
  SetHeaderContent,
} from "app/base/types";
import type {
  Machine,
  MachineEventErrors,
  SelectedMachines,
} from "app/store/machine/types";
import type { Script } from "app/store/script/types";

export type MachineHeaderContent = HeaderContent<
  ValueOf<typeof MachineHeaderViews>,
  {
    applyConfiguredNetworking?: Script["apply_configured_networking"];
    hardwareType?: HardwareType;
  }
>;

export type MachineSetHeaderContent = SetHeaderContent<MachineHeaderContent>;

export type MachineActionVariableProps = {
  machines?: Machine[];
  selectedMachines?: SelectedMachines | null;
  searchFilter?: string;
  selectedCount?: number | null;
  processingCount?: number;
  selectedCountLoading?: boolean;
};

export type MachineActionFormProps = Omit<
  CommonActionFormProps<MachineEventErrors>,
  "processingCount"
> &
  MachineActionVariableProps;
