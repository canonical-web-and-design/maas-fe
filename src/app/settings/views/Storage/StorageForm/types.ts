export type StorageFormValues = {
  default_storage_layout: string;
  disk_erase_with_quick_erase: boolean;
  disk_erase_with_secure_erase: boolean;
  enable_disk_erasing_on_release: boolean;
};
