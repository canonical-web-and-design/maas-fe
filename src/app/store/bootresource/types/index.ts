export type {
  DeleteImageParams,
  FetchParams,
  OsystemParam,
  SaveOtherParams,
  SaveUbuntuCoreParams,
  SaveUbuntuParams,
} from "./actions";
export type {
  BaseImageFields,
  BootResource,
  BootResourceEventError,
  BootResourceFetchedArch,
  BootResourceFetchedImages,
  BootResourceFetchedRelease,
  BootResourceOtherImage,
  BootResourceState,
  BootResourceStatuses,
  BootResourceUbuntu,
  BootResourceUbuntuArch,
  BootResourceUbuntuCoreImage,
  BootResourceUbuntuRelease,
  BootResourceUbuntuSource,
} from "./base";
export {
  BootResourceAction,
  BootResourceMeta,
  BootResourceSourceType,
  BootResourceType,
} from "./enum";
export type {
  BootResourceFetchResponse,
  BootResourcePollResponse,
} from "./responses";
