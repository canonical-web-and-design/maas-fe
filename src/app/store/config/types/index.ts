export type { Config, ConfigChoice, ConfigState, ConfigValues } from "./base";

export {
  AutoIpmiPrivilegeLevel,
  ConfigMeta,
  ConfigNames,
  NetworkDiscovery,
  TLSExpiryNotificationInterval,
} from "./enum";
