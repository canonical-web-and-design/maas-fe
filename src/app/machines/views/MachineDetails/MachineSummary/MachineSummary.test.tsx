import { mount } from "enzyme";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { CompatRouter, Route, Routes } from "react-router-dom-v5-compat";
import configureStore from "redux-mock-store";

import MachineSummary from "./MachineSummary";

import type { RootState } from "app/store/root/types";
import { NodeStatusCode } from "app/store/types/node";
import {
  machineDetails as machineDetailsFactory,
  machineState as machineStateFactory,
  rootState as rootStateFactory,
} from "testing/factories";

const mockStore = configureStore();

describe("MachineSummary", () => {
  let state: RootState;
  beforeEach(() => {
    state = rootStateFactory({
      machine: machineStateFactory({
        items: [machineDetailsFactory({ system_id: "abc123" })],
      }),
    });
  });

  it("displays a spinner if machines are loading", () => {
    state.machine.loading = true;
    const store = mockStore(state);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter
          initialEntries={[{ pathname: "/machine/abc123", key: "testKey" }]}
        >
          <CompatRouter>
            <MachineSummary setSidePanelContent={jest.fn()} />
          </CompatRouter>
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find("Spinner").exists()).toBe(true);
  });

  it("renders", () => {
    const store = mockStore(state);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter
          initialEntries={[{ pathname: "/machine/abc123", key: "testKey" }]}
        >
          <CompatRouter>
            <MachineSummary setSidePanelContent={jest.fn()} />
          </CompatRouter>
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find("MachineSummary")).toMatchSnapshot();
  });

  it("shows workload annotations for deployed machines", () => {
    state.machine.items = [
      machineDetailsFactory({
        status_code: NodeStatusCode.DEPLOYED,
        system_id: "abc123",
      }),
    ];
    const store = mockStore(state);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter
          initialEntries={[
            { pathname: "/machine/abc123/summary", key: "testKey" },
          ]}
        >
          <CompatRouter>
            <Routes>
              <Route
                element={<MachineSummary setSidePanelContent={jest.fn()} />}
                path="/machine/:id/summary"
              />
            </Routes>
          </CompatRouter>
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find("WorkloadCard").exists()).toBe(true);
  });

  it("shows workload annotations for allocated machines", () => {
    state.machine.items = [
      machineDetailsFactory({
        status_code: NodeStatusCode.ALLOCATED,
        system_id: "abc123",
      }),
    ];
    const store = mockStore(state);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter
          initialEntries={[
            { pathname: "/machine/abc123/summary", key: "testKey" },
          ]}
        >
          <CompatRouter>
            <Routes>
              <Route
                element={<MachineSummary setSidePanelContent={jest.fn()} />}
                path="/machine/:id/summary"
              />
            </Routes>
          </CompatRouter>
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find("WorkloadCard").exists()).toBe(true);
  });

  it("does not show workload annotations for machines that are neither deployed nor allocated", () => {
    state.machine.items = [
      machineDetailsFactory({
        status_code: NodeStatusCode.NEW,
        system_id: "abc123",
      }),
    ];
    const store = mockStore(state);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter
          initialEntries={[
            { pathname: "/machine/abc123/summary", key: "testKey" },
          ]}
        >
          <CompatRouter>
            <Routes>
              <Route
                element={<MachineSummary setSidePanelContent={jest.fn()} />}
                path="/machine/:id/summary"
              />
            </Routes>
          </CompatRouter>
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.find("WorkloadCard").exists()).toBe(false);
  });
});
