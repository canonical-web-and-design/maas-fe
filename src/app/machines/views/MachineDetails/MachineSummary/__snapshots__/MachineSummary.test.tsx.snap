// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MachineSummary renders 1`] = `
<MachineSummary
  setSidePanelContent={[MockFunction]}
>
  <Spinner
    text="Loading"
  >
    <span
      aria-live="polite"
      className="p-text--default"
      role="alert"
    >
      <i
        className="p-icon--spinner u-animation--spin"
      />
      â€‚
      <span>
        Loading
      </span>
    </span>
  </Spinner>
</MachineSummary>
`;
